<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="framerate" value="15" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name="gate_detector" pkg="gate_detector" type="gate_detector.py">
    <param name="_hue_lower" value="35" />
    <param name="_sat_lower" value="147" />
    <param name="_val_lower" value="98" />
    <param name="_hue_upper" value="56" />
    <param name="_sat_upper" value="255" />
    <param name="_val_upper" value="255" />
  </node>

  <node name="image_view0" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/gate_detector/image_detector"/>
    <param name="autosize" value="true" />
  </node>
    <node name="image_view1" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/gate_detector/hsv"/>
    <param name="autosize" value="true" />
  </node>
</launch>